app/GioForegroundService.java

```diff
@@ -0,0 +1,69 @@
+// SPDX-License-Identifier: Unlicense OR MIT
+
+package org.gioui;
+import android.app.Notification;
+import android.app.Service;
+import android.app.Notification;
+import android.app.Notification.Builder;
+import android.app.NotificationChannel;
+import android.app.NotificationManager;
+import android.app.PendingIntent;
+import android.content.Context;
+import android.content.Intent;
+import android.os.IBinder;
+
+public class GioForegroundService extends Service {
+	public static final String CHANNEL_ID = "GioForegroundService";
+	public static final String serviceName = "GioForeground";
+	public static final int notificationID = 42;
+
+	@Override public int onStartCommand (Intent intent, int flags, int startId) {
+		return START_NOT_STICKY;
+	}
+	@Override public IBinder onBind(Intent intent) {
+		return null;
+	}
+	@Override public void onCreate() {
+		super.onCreate();
+		this.createNotificationChannel();
+		this.startForeground();
+	}
+	public void startForeground() {
+		// create the Intent that will bring GioActivity to foreground
+		Context ctx = getApplicationContext();
+		Intent resultIntent = new Intent(ctx, GioActivity.class);
+		PendingIntent pending = PendingIntent.getActivity(ctx, notificationID, resultIntent, Intent.FLAG_ACTIVITY_CLEAR_TASK);
+		// create the Notification that Android will display on the tray for this service
+		Notification.Builder builder = new Notification.Builder(ctx, CHANNEL_ID)
+			.setContentTitle(CHANNEL_ID)
+			.setSmallIcon(getResources().getIdentifier("@mipmap/ic_launcher", "drawable", getPackageName()))
+			.setContentText(serviceName)
+			.setContentIntent(pending)
+			.setPriority(Notification.PRIORITY_DEFAULT);
+		startForeground(notificationID, builder.build());
+	}
+	public void stopForeground() {
+		stopForeground(true);
+	}
+	@Override
+	public void onTaskRemoved(Intent rootIntent) {
+		super.onTaskRemoved(rootIntent);
+		this.deleteNotificationChannel();
+		stopForeground(true);
+		this.stopSelf();
+	}
+	@Override public void onDestroy() {
+		this.deleteNotificationChannel();
+	}
+	private void deleteNotificationChannel() {
+		NotificationManager notificationManager = getSystemService(NotificationManager.class);
+		notificationManager.deleteNotificationChannel(CHANNEL_ID);
+	}
+	private void createNotificationChannel() {
+		// https://developer.android.com/training/notify-user/build-notification#java
+		NotificationChannel channel = new NotificationChannel(CHANNEL_ID, serviceName, NotificationManager.IMPORTANCE_DEFAULT);
```

Comment:
Why not a lower importance? The notification is only here because Android forces us to.

Reply:
ha you're telling me :P

> Oh, and remember to sign-off your change and squash commits.

Ok, I can do this and open a new PR once everything looks good.

Reply:
You can re-use this PR by force-pushing. For example:

$ git rebase -i main
$ git push --force-with-lease